cdisc_rules_engine/rules_engine.py

```diff
@@ -178,30 +185,41 @@ def get_dataset_builder(
             domain=domain,
             datasets=datasets,
             dataset_path=dataset_path,
+            define_metadata=self.define_metadata,
         )
 
     def validate_rule(
-        self, rule: dict, dataset_path: str, datasets: List[dict], domain: str
+        self,
+        rule: dict,
+        dataset_path: str,
+        datasets: List[dict],
+        domain: str,
     ) -> List[Union[dict, str]]:
         """
          This function is an entrypoint for rule validation.
         It defines a rule validator based on its type and calls it.
         """
+        kwargs = {}
         builder = self.get_dataset_builder(rule, dataset_path, datasets, domain)
         dataset = builder.get_dataset()
-        kwargs = {}
+        define_metadata = self.define_metadata
+
         # Update rule for certain rule types
         # SPECIAL CASES FOR RULE TYPES ###############################
         # TODO: Handle these special cases better.
         if (
```

Comment:
I think this logic should be done in the builder so that we can get rid of this section altogether and so that we don't need to pass around kwargs and define_metadata everywhere.
Refer to `define_item_group_dataset_builder`.
You could make the `domain` parameter optional in `get_define_xml_item_group_metadata`

Reply:
I created a method get_dataset_define_metadata in the base_dataset_builder, which does not use **kwargs. We check the rule type here. How would you like this part to be improved?

Reply:
I think this whole `if` branch should be removed. Any necessary logic should be inside the builder instead.
And I don't think we need `define_metadata`

Reply:
Removed the block.